cmd/execute.go

```diff
@@ -119,12 +121,26 @@ func execute() {
 	}
 
 	if len(ethStorageInitializers) > 0 {
-		tailer := fs.FileTailer{Path: storageDiffsPath}
-		storageFetcher := fetcher.NewCsvTailStorageFetcher(tailer)
-		sw := watcher.NewStorageWatcher(storageFetcher, &db)
-		sw.AddTransformers(ethStorageInitializers)
-		wg.Add(1)
-		go watchEthStorage(&sw, &wg)
+		switch stateDiffSource {
+		case "geth":
+			log.Debug("fetching storage diffs from geth pub sub")
+			rpcClient, _ := getClients()
+			stateDiffStreamer := streamer.NewStateDiffStreamer(rpcClient)
+			payloadChan := make(chan statediff.Payload)
+			storageFetcher := fetcher.NewGethRpcStorageFetcher(&stateDiffStreamer, payloadChan)
+			sw := watcher.NewGethStorageWatcher(&storageFetcher, &db)
```

Comment:
Why do you pass in the `storageFetcher` address to the `NewGethStorageWatcher` function but not to the `NewCsvStorageWatcher` function?

Reply:
Nice catch! This was required because the `GethRpcStorageFetcher`'s implementation of `FetchStorageDiffs` had a pointer receiver. But, I actually don't think that's necessary! Updated in 74f6f80.